Prevent duplicate link rel prefetches and prefetching the current page

nickFalcone opened the pull request:
Add a check against the prospective href to prefetch:

Is the href equal to the current page?

cramforce commented on src/main.js:
```diff
@@ -63,16 +63,24 @@ function message(msg) {
   if (e.target.origin != location.origin) {
     return;
   }
+  if (window.location.pathname === e.target.pathname) {
```
This would also need to compare the query string. So, all of the URL except for the fragment.

cramforce commented on src/main.js:
```diff
@@ -63,16 +63,24 @@ function message(msg) {
+  if (window.location.pathname === e.target.pathname) {
+    return;
+  }
+  if (prefetched.includes(e.target.href)) {
```
Please remove. The way the prefetch cache works is that it will cache the returning resource for 5 minutes independent of its caching headers and duplicate fetches should not happen. If this is not the case, please provide a DevTools screenshot showing it and ideally file a crbug.com

cramforce reviewed (changes requested):
Thanks for the PR! A couple comments.

nickFalcone commented:
Thanks for your feedback! 

I removed the check for duplicate prefetch and updated the comparison to include query string (excluding the URL fragment).

cramforce reviewed (approved):
Thanks!
